src/Functions/FunctionsStringDistance.cpp

```diff
@@ -211,10 +211,13 @@ struct ByteJaccardIndexImpl
             union_size = haystack_utf8_set.size() + needle_utf8_set.size() - intersection;
         }
 
-        for (size_t i = 0; i < max_size; ++i)
```

Comment:
Forget an `else` here.

Reply:
For utf-8 one-bytes code points are still stored in array. I assumed that some cases may benefit from it, when we call UTF8 version on ASCII string. I do not remember if improvement is significant, though.

https://github.com/ClickHouse/ClickHouse/blob/3b775dee5361a868facac76aafdd883e42b05b21/src/Functions/FunctionsStringDistance.cpp#L131